src/main/java/ca/corefacility/bioinformatics/irida/config/security/IridaWebSecurityConfig.java

```diff
@@ -197,26 +51,29 @@ public LoginSuccessHandler getLoginSuccessHandler() {
 
 		@Override
 		public void configure(WebSecurity web) throws Exception {
+			// @formatter:off
 			web.ignoring()
 					.antMatchers("/node_modules/**")
 					.antMatchers("/dist/**")
 					.antMatchers("/resources/**");
+
+			// @formatter:on
+			web.httpFirewall(new DefaultHttpFirewall());
 		}
 
 		@Override
 		protected void configure(HttpSecurity http) throws Exception {
 			authFailureHandler.setDefaultFailureUrl("/login?error=true");
 			// @formatter:off
 			http.requestMatcher(request -> {
-				return !request.getRequestURI().matches("^.*/api.*$");
+				return !request.getRequestURI().matches("^.*/api(?!/oauth/authorize|/oauth/authorization/token)(/.*)?$");
```

Comment:
I had to modify this regex to allow `/api/oauth/authorize` and `/api/oauth/authorization/token` to be handled by this WebSecurity config as we need the specific web security context that the login form gives us.